### PR TITLE
feat: expose scroll event

### PR DESCRIPTION
**Pull request by qili26**
1. Expose scroll event. Event can be used to check whether the scroll is at the bottom for endless data loading.

endless scrolling

### CODE REVIEW

**Comment by qili26** [comment]
https://github.com/adazzle/react-data-grid/pull/1854

**Comment by qili26** [comment]
https://github.com/adazzle/react-data-grid/issues/1804

File: src/DataGrid.tsx [hunk 1]
```diff
@@ -119,7 +119,7 @@ export interface DataGridProps<R, K extends keyof R, SR = unknown> {
   /** Function called whenever a row is clicked */
   onRowClick?: (rowIdx: number, row: R, column: CalculatedColumn<R, SR>) => void;
   /** Called when the grid is scrolled */
-  onScroll?: (scrollPosition: ScrollPosition) => void;
+  onScroll?: (scrollPosition: ScrollPosition, event: React.UIEvent<HTMLDivElement>) => void;
```
Comment (amanmahajan7) [inline src/DataGrid.tsx:122]: We can read `scrollPosition` from the event object so `onScroll(event)` can be used. Wdyt?

Comment (nstepien) [reply src/DataGrid.tsx:122]: I agree with @amanmahajan7, let's just pass the event only.

File: stories/demos/AllFeatures.tsx [hunk 2]
```diff
@@ -56,9 +56,27 @@ function createRows(numberOfRows: number): Row[] {
   return rows;
 }
 
+function isReachedBottom(event: React.UIEvent<HTMLDivElement>): boolean {
```
Comment (nstepien) [inline stories/demos/AllFeatures.tsx:59]: Small grammar fix.
```suggestion
function hasReachedBottom(event: React.UIEvent<HTMLDivElement>): boolean {
```
or
```suggestion
function isAtBottom(event: React.UIEvent<HTMLDivElement>): boolean {
```

File: stories/demos/AllFeatures.tsx [hunk 3]
```diff
@@ -190,26 +208,41 @@ export default function AllFeatures() {
     }
   }, []);
 
+  const handleScroll = useCallback(async (position: ScrollPosition, event: React.UIEvent<HTMLDivElement>) => {
```
Comment (nstepien) [inline stories/demos/AllFeatures.tsx:211]: Don't need `useCallback` here, changing `handleScroll` won't trigger any unnecessary re-renders.

File: stories/demos/AllFeatures.tsx [hunk 4]
```diff
@@ -190,26 +208,41 @@ export default function AllFeatures() {
...
+              rowHeight={30}
+              width={width}
+              height={height - 40}
+              selectedRows={selectedRows}
+              onScroll={handleScroll}
+              onSelectedRowsChange={setSelectedRows}
+              enableCellCopyPaste
+              enableCellDragAndDrop
+            />
+            {isLoading && <div className="load-more-rows-tag" style={{ '--load-more-rows-left': `${width - 230}px` } as React.CSSProperties}>Loading more rows...</div>}
```
Comment (nstepien) [inline stories/demos/AllFeatures.tsx:244]:
```suggestion
            {isLoading && <div className="load-more-rows-tag" style={{ left: width - 230 }}>Loading more rows...</div>}
```

File: stories/index.less [hunk 5]
```diff
@@ -11,3 +11,14 @@ body, html {
   height: 100vh;
   padding: 8px;
 }
+
+.load-more-rows-tag {
+  line-height: 35px;
+  width: 180px;
+  position: relative;
+  top: -70px;
+  left: var(--load-more-rows-left);
```
Comment (nstepien) [inline stories/index.less:20]:
```suggestion
```

File: src/DataGrid.tsx [hunk 6]
```diff
@@ -37,7 +37,6 @@ import {
   RowExpandToggleEvent,
   RowRendererProps,
   RowsUpdateEvent,
-  ScrollPosition,
```
Comment (nstepien) [inline src/DataGrid.tsx:40]: This type isn't used anywhere anymore, let's delete it.

Comment (qili26) [reply src/DataGrid.tsx:40]: Nice catch, I didn't know this detail.

**Pull request merged**